RPG Wizardry/Assets/Scripts/GameWorld/BackwallManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Tilemaps;
+
+namespace nl.SWEG.RPGWizardry.GameWorld
+{
+    [RequireComponent(typeof(Collider2D))]
+    public class BackwallManager : MonoBehaviour
+    {
+        #region Variables
+        /// <summary>
+        /// The filter that makes sure only the player is able to trigger the methods.
+        /// </summary>
+        [SerializeField]
+        private ContactFilter2D ContactFilter;
+
+        /// <summary>
+        /// The wall tilemaps.
+        /// </summary>
+        [Space]
+        [SerializeField]
+        private List<Tilemap> Walls;
+
+        /// <summary>
+        /// The trigger that checks if the player enters it.
+        /// </summary>
+        private Collider2D Collider;
+        #endregion
+
+        #region Methods
+        #region Private
+        /// <summary>
+        /// Uses Collider.OverlapCollider() to check if there are any Players or enemies in the collider.
+        /// </summary>
+        /// <returns>True if the collider is empty, false if it is not.</returns>
+        private bool GetOverlapData()
+        {
+            List<Collider2D> results = new List<Collider2D>();
+            return Collider.OverlapCollider(ContactFilter, results) >= 1;
+        }
+        #endregion
+
+        #region Unity
+        /// <summary>
+        /// Gets the collider.
+        /// </summary>
+        void Start()
+        {
+            Collider = GetComponent<Collider2D>();
+        }
+
+        /// <summary>
+        /// Checks if the thing entering the trigger is a player, and if it's fhe first thing to enter it, makes the walls transparent.
+        /// </summary>
+        /// <param name="collision">The thing entering the collider.</param>
+        private void OnTriggerEnter2D(Collider2D collision)
+        {
+            //Check if there is one or more things in the collider.
+            if (GetOverlapData())
+            {
+                //Set the alpha of each tilemap to half.
+                foreach (Tilemap s in Walls)

```

Comment:
Use a for-loop instead of a foreach-loop to prevent allocating memory for the loop